be/test/runtime/memory/memory_resource_test.cpp

```diff
@@ -0,0 +1,34 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#include "runtime/memory/memory_resource.h"
+
+#include <gtest/gtest.h>
+
+#include <vector>
+
+namespace starrocks {
+TEST(StackMemoryResource, Normal) {
+    // case allocate from stack
+    char buffer[1024];
+    stack_memory_resource mr(buffer, sizeof(buffer));
+    std::pmr::vector<char> res(&mr);
+    // allocate 16 bytes from stack
+    res.resize(4);
+    ASSERT_EQ((char*)res.data(), buffer);
+
+    // allocate
+    res.resize(128 / sizeof(char));
+    ASSERT_EQ((char*)res.data(), buffer + 4 * sizeof(char));
```

Comment:
Why start of res begin with 4 bytes beyond buffer?

Reply:
>     res.resize(4); 

allocate 4 bytes from stack. current offset is 4